File: CSharp/SpaceVIL/SpaceVIL/Core/ITextContainer.cs
using System.Drawing;

namespace SpaceVIL.Core
{
    /// <summary>
    /// An interface that describes texture for text rendering.
    /// </summary>
    public interface ITextContainer
    {
        /// <summary>
        /// Method for getting text texture.
        /// </summary>
        /// <returns>Texture as SpaceVIL.Core.ITextImage.</returns>
        ITextImage GetTexture();
        /// <summary>
        /// Method for getting text color.
        /// </summary>
        /// <returns>Text color as System.Drawing.Color.</returns>
        Color GetForeground();
    }
}

File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/Core/ITextContainer.java
package com.spvessel.spacevil.Core;

import java.awt.Color;

/**
 * An interface that describes texture for text rendering.
 */
public interface ITextContainer {
    /**
     * Method for getting text texture.
     * @return Texture as com.spvessel.spacevil.Core.ITextImage.
     */
    public ITextImage getTexture();

    /**
     * Method for getting text color.
     * @return Text color as java.awt.Color.
     */
    public Color getForeground();
}